File: easy/array/rotate_image.cpp
int m, n;

void solve(vector<vector<int> > &matrix){
	int col = matrix[0].size()-1;
	for(int i = 0; i < n/2; ++i){
		for(int j = 0; j < m; ++j){
			swap(matrix[j][i], matrix[j][col]); // swapping columns
		}
		col--;
	}
}

void transpose(vector<vector<int> > &matrix){
	for(int i = 0; i < m; ++i){
		for(int j = i; j < n; ++j){
			swap(matrix[i][j],matrix[j][i]);
		}
	}
}
	
void rotate(vector<vector<int>>&matrix) {
	m = matrix.size();
	n = matrix.size();
	transpose(matrix);
	solve(matrix);
}

File: easy/array/rotate_image.py
def solve(matrix):
	col = len(matrix[0])-1
	for i in range(n//2):
		for j in range(m):
			matrix[j][i], matrix[j][col] = matrix[j][col], matrix[j][i] # swapping columns
		col -= 1		

def transpose(matrix):
	for i in range(m):
		for j in range(i, n):
			matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

def rotate(matrix):
	global m, n
	m = len(matrix)
	n = len(matrix[0])
	print(n)
	transpose(matrix)
	solve(matrix)
	print(matrix)